app/views/projects/tree_filter/_blob_item.html.haml

```diff
@@ -0,0 +1,7 @@
+- tree_limit = 20 unless local_assigns[:tree_limit]
+
+%tr{ class: "tree-item #{blob_item_counter < tree_limit ? '' : 'hide'}" }
```

Comment:
We shouldn't render every single file, since this can run into the thousands upon thousands for large repos (the Linux tree has 50k+)